flask_resty/view.py

```diff
@@ -1082,19 +1108,17 @@ def upsert(self, id, *, with_for_update=False):
         :rtype: :py:class:`flask.Response`
         """
         data_in = self.get_request_data(expected_id=id)
+        item = self.get_item_or_none(id, with_for_update=with_for_update)
 
-        try:
-            item = self.get_item(id, with_for_update=with_for_update)
-        except NoResultFound:
-            item = self.create_and_add_item(data_in)
-            self.commit()
-
-            return self.make_created_response(item)
-        else:
-            item = self.update_item(item, data_in) or item
-            self.commit()
+        item = self.upsert_item(item, data_in) or item
+        updated = inspect(item).persistent
```

Comment:
this might not work if for some reason someone changes how the creation and upsert method work. what about upsert_item returns a tuple `(item: T, created: bool)`?

Reply:
I know. Awkward either way. It's why I'd be more inclined to add a new hook.